x-pack/plugins/cases/server/client/cases/create.ts

```diff
@@ -84,12 +88,17 @@ export const create = async ({
 
   try {
     try {
-      await auth.ensureAuthorized(query.owner, WriteOperations.Create);
+      await auth.ensureAuthorized(query.owner, Operations.createCase);
     } catch (error) {
-      // TODO: log error using audit logger
+      auditLogger?.log(createAuditMsg({ operation: Operations.createCase, error }));
       throw error;
     }
 
+    // log that we're attempting to create a case
+    auditLogger?.log(
```

Comment:
Should we audit log those? `logger` should be enough right?

Reply:
The Security Plugin's README seems to indicate that we need to log **read and write** operations and talks about logging intentions and the actual results:

https://github.com/elastic/kibana/blob/master/x-pack/plugins/security/README.md#what-events-should-be-logged

This is also how the alerting plugin does it.

Reply:
Yeh, this is correct, the approach we're taking is to log the intention for write operations. 

Please ensure that you log the ID of the case as well though so that it can be correlated. I don't know how cases are modelled but you can see an example of how we generate an ID for a saved object using `SavedObjectsUtils.generateId()` before writing it to Elasticsearch here: https://github.com/elastic/kibana/blob/4b4691b9b2f0e00648f2781ffe2a8162eb40a5d6/x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts#L101

Reply:
Ah got it. That makes sense. I'll generate the ID before the audit logger call and pass it into the saved object client's create().